### ISSUE TITLE
termusic failing to fill Library/Database

### ISSUE THREAD

**Issue by Porkepix** [tramhao/termusic#110]
I checked for a debug mode to try to check in logs what could happen. I installed termusic on a second laptop with a different Music directory, and noticed that besides the `mp3` directory listed (the one created by termusic itself, see https://github.com/tramhao/termusic/issues/109), it doesn't list any of the content of `~/Music`.

The origin of the install is the same (ArchLinux's AUR), though I could notice slight differences in configs. The one working correctly was installed with `0.7.7` and contains `["~/Music"]` for `music_dir`, while the laptop with the issue was installed with the `0.7.8` and contains `["/home/clement/Music/mp3", "/home/clement/Music"]` as `music_dir`  (note that one of them expanded `~` while the other didn't, I don't know why).

It could also be that some files in `~/Music` are giving termusic a hard time but I don't know why and a debug mode would be kind of helpful here I think.

**Comment by tramhao** [comment]
Yes, you're rignt. in v0.7.8 I expand the ~ when initialize the program, because win related problem reported for ~.  Debug mode is a great idea. I think I need to implement it. But as it's related to many aspect of the app, the implementation could be one module after another.

**Comment by tramhao** [comment]
The database is only updated for files below current root. If it's outside, you need to set it as root. However, the database will keep growing when you switch roots. 
It's quite confusing here. Probably some other implementation is better.

**Comment by Porkepix** [comment]
Well, I didn't used much the mode database on the laptop with no issue, I mostly used the library mode; on the laptop with the issue even in library mode there's nothing else than the `mp3` dir listed while `~/Music` contains many directories with Music, and I'm trying to figure out why but I guess something have termusic fails to populate files/directories. Could some files (archives etc.) result in a fail for termusic and have it interrupt the listing process?

**Comment by tramhao** [comment]
It's possible. I'm not so confident with the scan of database. Perhaps I add debug mode to this module first and see what's happening when sync database.

**Comment by Porkepix** [comment]
> It's possible. I'm not so confident with the scan of database. Perhaps I add debug mode to this module first and see what's happening when sync database.

Just to be clear, the issue I think arise even before populating the database: I guess the directories are first read and listed in the Library tab (`<1>`), and it's then used to populate the database. Here, even the library is shown as empty while it really isn't empty at all: 
![Screenshot from 2023-01-31 17-51-34](https://user-images.githubusercontent.com/709500/215828866-1a03849d-0378-4697-92a3-a518b622bae4.png)

I can't tell why, but only the created `mp3` directory is listed: every other directory is missing from here.

**Comment by tramhao** [comment]
Please use o key to switch root when you're focusing library. Currently the root is set as ~/Music/mp3, and that's why it's empty(because it's newly created)

**Comment by Porkepix** [comment]
Mmmh, okay, my bad there I didn't get that it worked that way (and it's a little bit confused to be honest).
I was tricked by the fact the other computer only had `["~/Music"]` in `music_dir` by default while the one I was reporting the issue for had ` ["/home/clement/Music/mp3", "/home/clement/Music"]` by default.

**Comment by tramhao** [comment]
That's another issue. From certain version, I use a configuration management tool, to merge old config and don't restore to default. Thus recently I change ~/Music to $XDG_AUDIO_DIR, but if you're running from an old config, the root value will not be changed.

**Comment by hasezoey** [comment]
could this issue be re-confirmed with latest master (currently at 3560924113989c6a17616ea2143c7f3d79fb87c9)?

since 0.7.11 the logger has been upgraded, many refactors have been done and some errors have improved.
the logger can now be enabled via `--log-to-file` which will output `/tmp/termusic-{server,tui}.log` files, or use environment variable `TM_LOGFILE=/path/to/tui.log` and `TMS_LOGFILE=/path/to/server.log`.

also to clarify the issue: this issue is about no files showing up in the `Music library` tree view, right?
to verify, what files are in the listed directory if you do `ls -al`?

**Comment by hasezoey** [comment]
> No, it actually switch roots with o very well back from and to it even if it doesn't exist. It's just counter-intuitive and have poor discoverability.

i think you misunderstood what i meant, i mean via `Backspace` going up a directory, not switching roots. (though in a sense switching roots of the tree, but not actually switching roots of music being played / what is listed in the config)
though i dont know if this is a actual issues, as i have not tried it yet, only my guess from reading previous comments.

> (does it edit the config file, or is it just for the current session?)

i dont know, as long as it modifies the `Settings` struct and the app exits gracefully (no crash or panic), then the config *should* get saved (the same as volume and things get saved)

**Comment by hasezoey** [comment]
for reference, the following is where it gets set:

https://github.com/tramhao/termusic/blob/3560924113989c6a17616ea2143c7f3d79fb87c9/lib/src/config/mod.rs#L41-L54

**Comment by tramhao** [comment]
Let me explain the history of this function.
In the beginning, there is only 1 root supported, thus it's always ~/Music.
Then there is a feature request for multiple roots. I added the function(not very nicely but works). In order to show how to configure it, by default it'll be 2 roots. 
And with this bug report, I decided not to create the folders, so there will be non-existing root folder. However, you can still move around with backspace, and set new root with enter.
I'm also a little confused about how to set a more reasonable config especially for new users. For old users, it'll be fine because as long as it's set once, it'll be saved to configuration.

**Comment by hasezoey** [comment]
slight update, e97fae72029182e8e311b12b98d468f8275cc98d has changed the order to have the `/mp3` directory *after* the normal music directory

**Comment by hasezoey** [comment]
> Also, it makes me think, does it handle well localized directories? (I can't tell, even though English isn't my native language, I set my OS to use English now).

no it does not, it uses a static `~/Music`, no XDG or similar used yet.
